index.bs

```diff
@@ -221,6 +221,24 @@ and doing cleanup as necessary (see the [[#listen-fully-active]] pattern above).
   but every feature specification should think about this and decide what works best for the feature.
 </div>
 
+<h4 id="non-synchronous-steps">Be aware that navigations can happen between steps running [=in parallel=]</h4>
+If steps that run [=in parallel=] queue a task that is associated with a document,
+you need to be aware that the document might have been navigated away from and (later on) navigated to again.
+This can happen while running the steps [=in parallel=] and can have happened when the steps in the queued task eventually run.
+Since a navigation has happened and a considerable amount of time might have passed between the steps,
+the user might not expect some states that are saved in the steps to be preserved.
+
+When writing steps that run [=in parallel=],
+consider also listening to changes to the [=Document/fully active=] state
+and doing cleanup as necessary or even refrain from running some steps
+(see the [[#listen-fully-active]] pattern above).
+<div class=example>
+  APIs that request permission to the user, or requests user interaction through prompt, etc.,
+  should consider re-requesting the user if it detects that the document had become non-fully active.
+</div>
+Note: In theory, steps that run [=in parallel=] ought to be already resilient to changes that might occur between its steps.
```

Comment:
Personally I would move this note up above the "When writing steps..." pattern.

Reply:
Moved now!